File: Tokopedia/Classses/Object/Shops/Notes/Notes.m
//
//  Notes.m
//  Tokopedia
//
//  Created by IT Tkpd on 10/10/14.
//  Copyright (c) 2014 TOKOPEDIA. All rights reserved.
//

#import "Notes.h"

@implementation Notes
+(RKObjectMapping *)mapping_v4{
    RKObjectMapping *statusMapping = [RKObjectMapping mappingForClass:[Notes class]];
    [statusMapping addAttributeMappingsFromDictionary:@{kTKPD_APISTATUSKEY:kTKPD_APISTATUSKEY,
                                                        kTKPD_APISERVERPROCESSTIMEKEY:kTKPD_APISERVERPROCESSTIMEKEY}];
    [statusMapping addPropertyMapping:[RKRelationshipMapping relationshipMappingFromKeyPath:@"data" toKeyPath:kTKPD_APIRESULTKEY withMapping:[NotesResult mapping]]];
    return statusMapping;
}
@end

File: Tokopedia/Classses/Object/Shops/Notes/Notes.swift
//
//  Notes.swift
//  Tokopedia
//
//  Created by Tedo Pranowo on 7/1/16.
//  Copyright © 2016 TOKOPEDIA. All rights reserved.
//

import Foundation

class Notes: NSObject {
    
    var status : NSString = ""
    var server_process_time : NSString = ""
    var result : NotesResult = NotesResult()
    
    
    class fileprivate func attributeMappingDictionary() -> [AnyHashable: Any]! {
        return ["status" : "status",
                "server_process_time" : "server_process_time"]
    }
    
    class func mapping() -> RKObjectMapping! {
        let mapping : RKObjectMapping = RKObjectMapping(for: self)
        
        mapping.addAttributeMappings(from:self.attributeMappingDictionary())
        mapping.addPropertyMapping(RKRelationshipMapping(fromKeyPath: "data", toKeyPath: "result", with: NotesResult.mapping()))
        
        return mapping
    }
}
